unlock-app/src/components/interface/pagination/PageNumbers.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { PageGroup, PageNumber, PageNumberActive } from './Pagination'
+import { PGN_MAX_NUMBER_OF_PAGES_TO_SHOW_ALL } from '../../../constants'
+
+const PageNumbers = ({ numberOfPages, currentPage, goToPage }) => {
+  const pageNumber = number => {
+    return number === currentPage ? (
+      <PageNumberActive
+        key={number.toString()}
+        onClick={() => goToPage(number)}
+      >
+        {number}
+      </PageNumberActive>
+    ) : (
+      <PageNumber key={number.toString()} onClick={() => goToPage(number)}>
+        {number}
+      </PageNumber>
+    )
+  }
+
+  // If number of pages less than 10, show all 10 page numbers
+  if (numberOfPages <= PGN_MAX_NUMBER_OF_PAGES_TO_SHOW_ALL) {
+    return (
+      <PageGroup>
+        {Array(numberOfPages)
+          .fill()
+          .map((_, pn) => {
+            return pageNumber(pn + 1)
+          })}
+      </PageGroup>
+    )
```

Comment:
The code:

```js
Array(numberOfPages)
          .fill()
          .map((_, pn) => {
            return pageNumber(pn + 1)
          })
```

is used quite a bit in this component. Perhaps we can do this once, and use slices, so for example on line 41:

```js
{Allpages.slice(0, 5)}
```

and a similar usage on line 81 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice